### ISSUE TITLE
Rule Proposal: Always use :focus when using :hover

### ISSUE THREAD

**Issue by nzakas** [CSSLint/csslint#264]
For accessibility purposes, it's usually advisable to include a rule for :focus whenever there's something for :hover. So

``` css
a:hover,
a:focus {
    color:red;
}
```

**Comment by mgifford** [comment]
Wow, been over 4 years now and nobody else has stepped forward on this yet.  I'm still interested. 

@stubbornella do you have any suggestions on getting started? I'm more of a PHP guy. 

So for a simple test, I went here http://csslint.net/

I would like to get an error when I do this:
```
a:hover {
    color:red;
}
```

but not when you provide the example that @nzakas did for this back in 2012. 

I know I talked to @nschonni about this in the past. 

This all goes back to [WCAG 2.0 AA Success Criterion 2.4.7]( https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-visible.html)

So how much more detail is needed in the Rule? 
  https://github.com/CSSLint/csslint/wiki/Rules

**Comment by mgifford** [comment]
@Andrew Macpherson https://github.com/fuzzbomb in a Drupal slack chat suggested maybe focus-within is [more appropriate in some instances](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-within):

```
tr:hover,
tr:focus-within {}
```

Maybe a linter can say "whenever there's an `a:hover`, there should also be a `a:focus` but they don't have to use the same styles." Like this:

```
a:hover { ... subtle style ...}
a:focus { ... strong style ... }
```

**Comment by mgifford** [comment]
Here's the WCAG 2.1 guideline https://www.w3.org/TR/WCAG21/#content-on-hover-or-focus

**Comment by stubbornella** [comment]
I think your approach makes sense. I’d find a similar rule (one that tests
properties and values, not just selectors). Copy that rule. Modify the
tests. And then iterate toward it testing the right thing. Lemme know how
it goes!
